### PR TITLE
Remove deprecated methods

### PR DESCRIPTION
**Pull request by ryanking13**
Remove deprecated methods before releasing 0.23 following the [deprecation timeline](https://github.com/pyodide/pyodide/blob/main/docs/project/deprecation-timeline.md)

### CODE REVIEW

**Comment by hoodmane** [comment]
Thanks @ryanking13! You need to keep
```py
from . import _state # noqa: F401
```
in `__init__.py`

**Comment by ryanking13** [comment]
Good point, thanks!

File: src/tests/test_pyodide_http.py [hunk 1]
```diff
@@ -12,8 +12,8 @@ def test_open_url(selenium, httpserver):
     assert (
         selenium.run(
             f"""
-            import pyodide
-            pyodide.open_url('{request_url}').read()
+            from pyodide.http import open_url
```
Comment (hoodmane) [inline src/tests/test_pyodide_http.py:15]: Have to add
```js
    pyodide.pyimport("pyodide.http").destroy();
```
to `INITIALIZE_SCRIPT`.

**Review by rth — APPROVED** [review]
Maybe we should still mention it in the changelog that deprecated methods were removed?
Otherwise LGTM.

**Pull request merged**